### ISSUE TITLE
[FEA] Support reading delta_log files on GPU

### ISSUE THREAD

**Issue by tgravescs** [NVIDIA/spark-rapids#8048]
**Is your feature request related to a problem? Please describe.**
Currently we don't support reading the metadata delta_log json files or checkpoint.parquet files.  For those we fallback to the CPU as its more efficient. We should look into running those on the GPU.

There are multiple different things that happen:

Checkpoint file (parquet)
1) parquet file reads
2) Delta Table Checkpoint with Stats #117282 - s3://.../_delta_log [34] [Cached]
map at StateCache.scala:55, I thought this was run as a UDF, needs investigation

json files:
1) json file reas
2) possibly UDF
3)  we have seen objecthashaggregate in here as well
4) UnionWithLocalData

**Comment by tgravescs** [comment]
one of the biggest things for the delta_log json files is having a performant and accurate json parser.

**Comment by jlowe** [comment]
> one of the biggest things for the delta_log json files is having a performant and accurate json parser

Agreed.  Given the relatively small amount of data in each JSON file, we may need to look into coalescing files to get a bigger amount of data for the GPU to process per invocation.

We also probably need to cover the rest of the metadata query on the GPU.  Unless the JSON parser is _really_ fast compared to the CPU (unlikely unless there is a ton of JSON to process), there's immediately going to be a fallback to the CPU for the very next operation (e.g.: ObjectHashAggregate, MapPartitions, etc.) and that transition will likely kill the speed gains from the JSON parsing on the GPU.

There's also the possibility that the JSON parsing has been cached, and the metadata query will start with an ExstingRDD Scan, and that may be tricky to speed up given the initial CPU rows -> GPU columns conversion at the very start.

**Comment by tgravescs** [comment]
Projects fallback because:

```
!Expression <ScalaUDF> UDF(add#521011.path) cannot run on GPU because neither UDF implemented by class org.apache.spark.sql.delta.DeltaLog$Canonical
PathFunction provides a GPU implementation, nor the conf `spark.rapids.sql.rowBasedUDF.enabled` is enabled
```

**Comment by tgravescs** [comment]
this would need https://github.com/NVIDIA/spark-rapids/issues/14465
